.buildkite/scripts/resolve_es_branch.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+set -e
+
+VERSION_URL="https://raw.githubusercontent.com/elastic/logstash/main/ci/logstash_releases.json"
+
+echo "Fetching versions from $VERSION_URL"
+VERSIONS=$(curl --silent $VERSION_URL)
+
+if [[ "$SNAPSHOT" == "true" ]]; then
+  key=$(echo "$VERSIONS" | jq '.snapshots."'"$ELASTIC_STACK_VERSION"'"')
+else
+  key=$(echo "$VERSIONS" | jq '.releases."'"$ELASTIC_STACK_VERSION"'"')
+fi
+
+# remove the wrapped quotation marks ("")
+resolved_stack_version=$(echo "$key" | sed 's/\"//g')
+
+IFS='.'
+read -a versions <<< "$resolved_stack_version"
+target_branch=${versions[0]}.${versions[1]}
```

Comment:
getting the major and minor version sounds like a good candidate for a [bash PE expression](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html) i.e. replace all the three lines above with:

`target_branch=${resolved_stack_version%.*}`

this works from right to left and removes every character up to and including the first found `.`:

```
$ echo $key
8.12.0-SNAPSHOT

$ echo ${key%.*}
8.12
```